xlr/sdk/src/sdk.ts

```diff
@@ -299,49 +302,93 @@ export class XLRSDK {
     throw new Error(`Unknown export format ${exportType}`);
   }
 
-  private resolveType(type: NodeType): NamedType {
-    return simpleTransformGenerator("object", "any", (objectNode) => {
-      if (objectNode.extends) {
-        const refName = objectNode.extends.ref.split("<")[0];
-        let extendedType = this.getType(refName, { getRawType: true });
-        if (!extendedType) {
-          throw new Error(
-            `Error resolving ${objectNode.name}: can't find extended type ${refName}`
-          );
-        }
+  /**
+   * Transforms a generated XLR node into its final representation by
+   * Resolving all `extends` properties
+   * Solving any conditional types
+   * Computing the effective types of any union elements
+   * Resolving any ref nodes
+   * filing in any remaining generics with their default value
+   */
+  private resolveType(type: NamedType, optimize = true): NamedType {
```

Comment:
i know this is a private method, but i think JSDoc would still be good

like what does optimization mean here?

Reply:
Good point. That behavior is documented on the public API that is the one caller of this method but I’ll add some more info just incase for the future something else leverages it.